R/label_operators.R

```diff
@@ -1,3 +1,48 @@
+#' Apply Labels to a dataset
+#'
+#' This function adds "label" attributes to all columns in a dataset
+#' Using the labels_app.csv file which contains common labels for ADPP
+#' and ADPC column names
+#'
+#' @param data The dataset to which labels will be applied.
+#'
+#' @return The same dataset with label attributes applied to all columns.
+#' If a column is not present in the labels list, it will be assigned the name of the col.
+#'
+#' @examples
+#' \dontrun{
+#'   # Example usage:
+#'   data <- data.frame(USUBJID = c(1, 2, 3), AVAL = c(4, 5, 6))
+#'   data <- apply_labels(data)
+#'   print(attr(data$A, "label"))
+#' }
+#'
+#' @export
+apply_labels <- function(data) {
+
+  labels_app <- read.csv(system.file("shiny/www/data/labels_app.csv", package = "aNCA"))
```

Comment:
issue: If possible, we should avoid loading files inside of functions - this means the file will be loaded everytime the function is called (which should not be a lot in the app, nevertheless this is bad practice).

I would suggest loading the specification file on app-level, and then taking the data frame as an argument. This will also make the function more versatile - we could load different specifications with it, theoretically.